pkg/workloadmeta/collectors/internal/containerd/image_sbom_trivy.go

```diff
@@ -65,18 +65,49 @@ func (c *collector) startSBOMCollection(ctx context.Context) error {
 						continue
 					}
 
-					if err := c.extractBOMWithTrivy(ctx, image); err != nil {
+					if err := c.extractBOMWithTrivy(ctx, image, ch); err != nil {
 						log.Warnf("Error extracting SBOM for image: namespace=%s name=%s, err: %s", image.Namespace, image.Name, err)
 					}
 				}
 			}
 		}
 	}()
 
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+			case result := <-ch:
+				if result.Error != nil {
+					log.Errorf("Failed to generate SBOM for containerd image: %s", result.Error)
+					continue
+				}
+
+				bom, err := result.Report.ToCycloneDX()
+				if err != nil {
+					log.Errorf("Failed to extract SBOM from report")
+					continue
+				}
+
+				sbom := workloadmeta.SBOM{
+					CycloneDXBOM:       bom,
+					GenerationTime:     result.CreatedAt,
+					GenerationDuration: result.Duration,
+				}
+
+				// Updating workloadmeta entities directly is not thread-safe, that's why we
+				// generate an update event here instead.
+				if err := c.handleImageCreateOrUpdate(ctx, result.ImgMeta.Namespace, result.ImgMeta.Name, &sbom); err != nil {
+					log.Warnf("Error extracting SBOM for image: namespace=%s name=%s, err: %s", result.ImgMeta.Namespace, result.ImgMeta.Name, err)
+				}
+			}
+		}
+	}()
+
 	return nil
 }
 
-func (c *collector) extractBOMWithTrivy(ctx context.Context, storedImage *workloadmeta.ContainerImageMetadata) error {
+func (c *collector) extractBOMWithTrivy(ctx context.Context, storedImage *workloadmeta.ContainerImageMetadata, ch chan sbom.ScanResult) error {
```

Comment:
you can make it more explicit that this function with push the result into the chan with 
```suggestion
func (c *collector) extractBOMWithTrivy(ctx context.Context, storedImage *workloadmeta.ContainerImageMetadata, ch chan<- sbom.ScanResult) error {
```

Reply:
You can also rename `ch` to a more explicit name, maybe `resultChan`